hnn_core/network.py

```diff
@@ -832,6 +861,254 @@ def _get_src_type_and_pos(self, gid):
+
+        src_cell = 'L5Pyr'
+        nc_dict['lamtha'] = 3.
+        key = f'gbar_L5Pyr_{target_cell}'
+        nc_dict['A_weight'] = self.params[key]
+        loc = 'soma'
+        receptor = 'ampa'
+        self._all_to_all_connect(
+            connectivity, src_cell, target_cell, loc, receptor, nc_dict)
+
+        src_cell = 'L2Pyr'
+        key = f'gbar_L2Pyr_{target_cell}'
+        nc_dict['A_weight'] = self.params[key]
+        loc = 'soma'
+        receptor = 'ampa'
+        self._all_to_all_connect(
+            connectivity, src_cell, target_cell, loc, receptor, nc_dict)
+
+    def _all_to_all_connect(self, connectivity_list, src_cell, target_cell,
+                            loc, receptor, nc_dict,
+                            allow_autapses=True, unique=False):
+        """Generate connectivity list given lists of sources and targets.
+
+        Parameters
+        ----------
+        connectivity_list : list of dict
+            The connectivity list to be updated
+        src_cell : str
+            Source cell type.
+        target_cell : str
+            Target cell type.
+        loc : str
+            If 'proximal' or 'distal', the corresponding
+            dendritic sections from Cell.sect_loc['proximal']
+            or Cell.Sect_loc['distal'] are used
+        receptor : str
+            The receptor.
+        nc_dict : dict
+            The connection dictionary containing keys
+            A_delay, A_weight, lamtha, and threshold.
+        allow_autapses : bool
+            If True, allow connecting neuron to itself.
+        unique : bool
+            If True, each target cell gets one "unique" feed.
+            If False, all src_type cells are connected to
+            all target_type cells.
+        """
+        connectivity = list()
+
+        src_gids = self.gid_ranges[_long_name(src_cell)]
+        target_gids = self.gid_ranges[_long_name(target_cell)]
+        src_types = np.repeat(src_cell, len(src_gids))
+        target_types = np.repeat(target_cell, len(target_gids))
+
+        src_start = src_gids[0]  # Necessary for unique feeds
+        for target_type, target_gid in zip(target_types, target_gids):
+            if unique:
+                src_gids = [target_gid + src_start]
+                src_types = [src_types[0]]  # Assumes that all src_types match
+            for src_type, src_gid in zip(src_types, src_gids):
+                conn = {'src_type': src_type, 'src_gid': src_gid,
+                        'target_type': target_type, 'target_gid': target_gid,
+                        'loc': loc, 'receptor': receptor,
+                        'nc_dict': deepcopy(nc_dict)}
+                if not allow_autapses and src_gid == target_gid:
+                    continue
+
+                connectivity.append(conn)
+        connectivity_list.extend(connectivity)
+
+    def add_connection(self, src_gid, target_gid, loc, receptor, delay,
+                       weight, lamtha, threshold=0.0):
```

Comment:
```suggestion
                       weight, lamtha):
```

is there a use case to support different thresholds for different connections?

Reply:
exposing more arguments later is not hard. Removing is hard. I'd suggest to start with fewer and add later depending on requirement

Reply:
Agreed, I'll remove threshold. However, I think it is still useful to expose the lamtha parameter as it is varied in the default instantiation of the connectivity.